packages/components/src/List/ListComposition/SortBy/SortBy.component.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Navbar, NavDropdown, Nav, NavItem, MenuItem } from 'react-bootstrap';
+import uuid from 'uuid';
+import isUndefined from 'lodash/isUndefined';
+
+import { useListContext } from '../context';
+
+function SortBy(props) {
+	const { id, options, selected, isDescending: isDescendingProp, onChange, onOrderChange } = props;
+	const { sortParams, setSortParams, t } = useListContext();
+
+	// Sort by
+	let selectedOption;
+	if (selected) {
+		selectedOption = selected;
+	} else if (sortParams.sortBy) {
+		selectedOption = sortParams.sortBy;
+	} else {
+		selectedOption = options[0].key;
+
+		useEffect(() => {
```

Comment:
conditional hooks are forbidden

Reply:
Oops correct
*Don’t call Hooks inside loops, conditions, or nested functions*
Linter didn't catch me though, will fix it.